commands/proposal_submission.go

```diff
@@ -572,19 +572,32 @@ func checkDataSourceSpec(spec *vegapb.DataSourceDefinition, name string, parentP
 
 	switch spec.SourceType.(type) {
 	case *vegapb.DataSourceDefinition_External:
-		signers := spec.GetSigners()
-		filters := spec.GetFilters()
-		if isBuiltInSpec(filters) {
-			return checkDataSourceSpecFilters(filters, name, parentProperty)
-		}
-		if len(signers) == 0 {
-			errs.AddForProperty(fmt.Sprintf("%s.%s.signers", parentProperty, name), ErrIsRequired)
-		}
-		for i, key := range signers {
-			signer := types.SignerFromProto(key)
-			if signer.IsEmpty() {
-				errs.AddForProperty(fmt.Sprintf("%s.%s.signers.%d", parentProperty, name, i), ErrIsNotValid)
-			}
+	  // For now check only for oracle type for external data source. Add a check for Oracle type data later when other sources are added.
+	  o := tp.External.GetOracle()
+  
+	  signers := o.Signers
+	  if len(signers) == 0 {
+		  errs.AddForProperty(fmt.Sprintf("%s.%s.external.oracle.signers", parentProperty, name), ErrIsRequired)
+	  }
+  
+	  for i, key := range signers {
+		  signer := types.SignerFromProto(key)
+		  if signer.IsEmpty() {
+			  errs.AddForProperty(fmt.Sprintf("%s.%s.external.oracle.signers.%d", parentProperty, name, i), ErrIsNotValid)
+		  }
+	  }
+	  
+	  filters := o.Filters
+	  if isBuiltInSpec(filters) {
+		  return checkDataSourceSpecFilters(filters, fmt.Sprintf("external.oracle.%s", name), parentProperty)
+	  }
+  
+	  errs.Merge(checkDataSourceSpecFilters(filters, fmt.Sprintf("external.oracle.%s", name), parentProperty))
+	  }
+	}
+	errs.Merge(checkDataSourceSpecFilters(filters, name, parentProperty))
+		errs.Merge(checkDataSourceSpecFilters(filters, name, parentProperty))
+	}
```

Comment:
```
errs.Merge(checkDataSourceSpecFilters(filters, name, parentProperty))
		errs.Merge(checkDataSourceSpecFilters(filters, name, parentProperty))
	}
```
This last merge should be deleted I think, because we do it inside the `if` and do not have access to filters anymore here